core/src/main/java/software/amazon/awssdk/core/http/loader/DefaultSdkHttpClientFactory.java

```diff
@@ -17,14 +17,13 @@
 
 import software.amazon.awssdk.core.exception.SdkClientException;
 import software.amazon.awssdk.http.SdkHttpClient;
-import software.amazon.awssdk.http.SdkHttpClientFactory;
 import software.amazon.awssdk.http.SdkHttpService;
 import software.amazon.awssdk.utils.AttributeMap;
 
 /**
  * Utility to load the default HTTP client factory and create an instance of {@link SdkHttpClient}.
  */
-public final class DefaultSdkHttpClientFactory implements SdkHttpClientFactory {
+public final class DefaultSdkHttpClientFactory implements SdkHttpClient.Builder {
```

Comment:
seems like there are some inconsistency between async and sync class names.  `DefaultBuilder` vs. `DefaultSdkHttpClientFactory`, I'd vote for `DefaultSdkAsyncHttpClientBuilder`, `DefaultSdkHttpClientBuilder`